remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java

```diff
@@ -188,24 +188,20 @@ public CollectionWithPagingInfo<Tag> getTags(StoreRef storeRef, Parameters param
 
     public NodeRef validateTag(String tagId)
     {
-    	NodeRef tagNodeRef = nodes.validateNode(tagId);
-    	if(tagNodeRef == null)
-    	{
-    		throw new EntityNotFoundException(tagId);
-    	}
-    	return tagNodeRef;
-    }
+		NodeRef tagNodeRef = nodes.validateNode(tagId);
+		return validateTag(tagNodeRef.getStoreRef(), tagId);
+	}
     
     public NodeRef validateTag(StoreRef storeRef, String tagId)
     {
-    	NodeRef tagNodeRef = nodes.validateNode(storeRef, tagId);
-    	if(tagNodeRef == null)
-    	{
-    		throw new EntityNotFoundException(tagId);
-    	}
-    	return tagNodeRef;
-    }
-
+		NodeRef tagNodeRef = nodes.validateNode(storeRef,tagId);
+		String tagName = taggingService.getTagName(tagNodeRef);
+		if( tagNodeRef == null || !taggingService.isTag(tagNodeRef.getStoreRef(), tagName))
```

Comment:
Can we improve this? Currently we are doing the following:
* Check there is a node for `storeRef` and `tagId` and get the `tagNodeRef`.
* Use `tagNodeRef` to get the name of that node, ignoring whether it's actually a tag or not.
* Check if `tagNodeRef` was null.
* Get the store ref from `tagNodeRef`.
* Use search to check if there's a node with the `cm:taggable` aspect and the given `tagName`.

This is pretty complex and I don't even think this works properly as it is. If we have a tag called "apples" and call this method with a node id for a folder called "apples" then I think it will say the folder is a valid tag.

Reply:
Addressed review comment